### PR TITLE
Unify all methods

### PR DESCRIPTION
**Pull request by fedetask**
Adding cmd arguments
- `-sa`: Use Simulated Annealing. If not present, use standard JaBeJa.
- `-factor`: Simulated Annealing factor.

### CODE REVIEW

File: run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+./run.sh -graph=graphs/twitter.graph -temp=1.0 -sa -delta=0.99 -factor=1.0
+
+# Experiments with different factors
+./run.sh -graph=graphs/3elt.graph -temp=1.0 -sa -delta=0.9 -factor=0.5
+./run.sh -graph=graphs/add20.graph -temp=1.0 -sa -delta=0.9 -factor=0.5
+./run.sh -graph=graphs/twitter.graph -temp=1.0 -sa -delta=0.9 -factor=0.5
+
+./run.sh -graph=graphs/3elt.graph -temp=1.0 -sa -delta=0.9 -factor=1.5
+./run.sh -graph=graphs/add20.graph -temp=1.0 -sa -delta=0.9 -factor=1.5
+./run.sh -graph=graphs/twitter.graph -temp=1.0 -sa -delta=0.9 -factor=1.5
```
Comment (OleguerCanal) [inline run_tests.sh:42]: Nice!!!

File: src/main/java/se/kth/jabeja/Jabeja.java [hunk 2]
```diff
@@ -41,19 +52,20 @@ public void startJabeja() throws IOException {
...
       //reduce the temperature
       saCoolDown();
-      report();
+      // Restart
+      if (round%300 == 0) {
+        T = config.getTemperature();
+      }
+      report(true, false);  // Save, do not print
     }
+    report(false, true);  // Do not save, print
```
Comment (OleguerCanal) [inline src/main/java/se/kth/jabeja/Jabeja.java:61]: Nice

File: src/main/java/se/kth/jabeja/Jabeja.java [hunk 3]
```diff
@@ -41,19 +52,20 @@ public void startJabeja() throws IOException {
       //one cycle for all nodes have completed.
       //reduce the temperature
       saCoolDown();
```
Comment (OleguerCanal) [inline src/main/java/se/kth/jabeja/Jabeja.java:54]: Should we call this even when not running SA? I thought we then wanted linear evolution of T

Comment (fedetask) [reply src/main/java/se/kth/jabeja/Jabeja.java:54]: You're right!

File: src/main/java/se/kth/jabeja/Jabeja.java [hunk 4]
```diff
@@ -65,7 +65,10 @@ public void startJabeja() throws IOException {
    * Simulated analealing cooling function
    */
   private void saCoolDown(){
-    T = Math.max(T*config.getDelta(), T_min);
+    if (simulatedAnnealing)
+      T = Math.max(T*config.getDelta(), T_min);
+    else
+      T = Math.max(T - config.getDelta(), T_min);
```
Comment (OleguerCanal) [inline src/main/java/se/kth/jabeja/Jabeja.java:71]: nice

**Review by OleguerCanal — APPROVED** [review]

**Pull request merged**